cirq-ionq/cirq_ionq/ionq_native_target_gateset_test.py

```diff
@@ -0,0 +1,712 @@
+
+
+# test init and atol argument
+def test_AriaNativeGateset_init():
+    gateset = AriaNativeGateset(atol=7)
+    assert gateset.atol == pytest.approx(7)
+
+
+# test init and atol argument
+def test_ForteNativeGateset_init():
+    gateset = ForteNativeGateset(atol=7)
+    assert gateset.atol == pytest.approx(7)
+
+
+# test _value_equality_values_ method
+def test_AriaNativeGateset__value_equality_values_():
+    gateset = AriaNativeGateset(atol=7)
+    assert gateset._value_equality_values_() == pytest.approx(7)
+
+
+# test _value_equality_values_ method
+def test_ForteNativeGateset_value_equality_values_():
+    gateset = ForteNativeGateset(atol=7)
+    assert gateset._value_equality_values_() == pytest.approx(7)
+
+
+# test _json_dict_ method
+def test_AriaNativeGateset__json_dict_():
+    gateset = AriaNativeGateset(atol=7)
+    assert str(gateset._json_dict_()) == "{'atol': 7}"
+
+
+# test _json_dict_ method
+def test_ForteNativeGateset__json_dict_():
+    gateset = ForteNativeGateset(atol=7)
+    assert str(gateset._json_dict_()) == "{'atol': 7}"
+
+
+# test _from_json_dict_ method
+def test_AriaNativeGateset__from_json_dict():
+    gateset = AriaNativeGateset(atol=7)
+    assert repr(gateset._from_json_dict_(7)) == "cirq_ionq.AriaNativeGateset(atol=7)"
+
+
+# test _from_json_dict_ method
+def test_ForteNativeGateset__from_json_dict():
+    gateset = ForteNativeGateset(atol=7)
+    assert repr(gateset._from_json_dict_(7)) == "cirq_ionq.ForteNativeGateset(atol=7)"
+
+
+# test _decompose_two_qubit_operation on non unitary argument
+def test_AriaNativeGateset_decompose_two_qubit_operation():
+    gateset = AriaNativeGateset(atol=7)
+    result = gateset._decompose_two_qubit_operation(
+        cirq.MeasurementGate(num_qubits=1, key='key'), "blank"
+    )
+    assert result == NotImplemented
+
+
+# test _decompose_two_qubit_operation on non unitary argument
+def test_ForteNativeGateset_decompose_two_qubit_operation():
+    gateset = ForteNativeGateset(atol=7)
+    result = gateset._decompose_two_qubit_operation(
+        cirq.MeasurementGate(num_qubits=1, key='key'), "blank"
+    )
+    assert result == NotImplemented
+
+
+# test CCZ_gate not working with 2 qubits
+def test_CCZ_gate_not_working_with_2_qubits():
+    with pytest.raises(Exception) as exc_info:
+        gateset = AriaNativeGateset()
+        gateset.decompose_all_to_all_connect_ccz_gate(cirq.CCZ, cirq.LineQubit.range(2))
+    assert exc_info.value.args[0] == 'Expect 3 qubits for CCZ gate, got 2 qubits.'
+
+
+# test CCZ_gate not working with 1 qubits
+def test_CCZ_gate_not_working_with_1_qubit():
+    with pytest.raises(Exception) as exc_info:
+        gateset = ForteNativeGateset()
+        gateset.decompose_all_to_all_connect_ccz_gate(cirq.CCZ, cirq.LineQubit.range(1))
+    assert exc_info.value.args[0] == 'Expect 3 qubits for CCZ gate, got 1 qubits.'
+
+
+# Tests for transpiling one qubit circuits
+qubit1 = cirq.LineQubit.range(2)
```

Comment:
Nit - instead of defining qubit1, qubits2, ... you can define the array just once as qubits = ...LineQubit.range(3).
There is also a builtin `cirq.q` for that, e.g., using `_q = cirq.q` and then get the qubits with `_q(0)`, `_q(1)`, etc.

Reply:
renamed 1qubit to oneQubitLine, 2qubits to twoQubitsLine etc.